### ISSUE TITLE
Dropdown component - screen readers not announcing dropdown item, once custom ID for item is defined

### ISSUE THREAD

**Issue by aubreyquinn** [microsoft/fluentui#19621]
### Environment Information

- **Package version(s)**: 0.54.0
- **Browser and OS versions**: Edge & Windows 10

### Describe the issue:

When I use JAWS screen reader or NVDA, it does not announce the header. 

Enter a search term into the text box. e.g. "a".
Use the up and down arrows to navigate the results.
Note that it does not announce the header of the result, but rather the search term "a".

Code sandbox: https://codesandbox.io/s/fluent-ui-example-drop-down-complex-object-65n4r?file=/example.js

#### Actual behavior:

Enter a search term into the text box. e.g. "a".
Use the up and down arrows to navigate the results.
Note that it does not announce the header of the result, but rather the search term "a".

When I remove the "id" field from the object, the screen readers work.
e.g.
```
  {
   // id: 1,
    header: "Steven Strange",
    image: "abc",
    content: "sstrange@microsoft.com"
  },
```

#### Expected behavior:

Enter a search term into the text box. e.g. "a".
Use the up and down arrows to navigate the results.
The screen reader should announce the header text.

### Documentation describing expected behavior

https://fluentsite.z22.web.core.windows.net/0.54.0/components/dropdown/definition

**Comment by gouttierre** [comment]
@aubreyquinn - thanks for filing this issue with us.

@jurokapsiar -  Would your team be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by aubreyquinn** [comment]
@jurokapsiar yes, it's still an issue for v 0.54
